test/cases/ubi-wsl.sh

```diff
@@ -192,20 +192,33 @@ $AZURE_CMD vm create \
    --attach-os-disk "$AZ_DISK" \
    --os-type "windows" \
    --security-type "TrustedLaunch" \
+   --public-ip-sku Standard \
    --location "$AZURE_WSL_LOCATION" \
     --nic-delete-option delete \
     --os-disk-delete-option delete \
     --size "Standard_D2as_v5"
 
 $AZURE_CMD vm open-port --resource-group "$AZURE_RESOURCE_GROUP" --name "wsl-vm-$TEST_ID" --port 22
-HOST=$($AZURE_CMD vm list-ip-addresses \
-   --resource-group "$AZURE_RESOURCE_GROUP" \
-   --name "wsl-vm-$TEST_ID" \
-   --query "[].virtualMachine.network.publicIpAddresses[0].ipAddress" \
-   --output tsv)
 
-greenprint "🛃 Wait until sshd is up"
+greenprint "🛃 Wait until the VM has a public IP"
+for LOOP_COUNTER in {0..30}; do
+    HOST=$($AZURE_CMD vm list-ip-addresses \
+       --resource-group "$AZURE_RESOURCE_GROUP" \
+       --name "wsl-vm-$TEST_ID" \
+       --query "[].virtualMachine.network.publicIpAddresses[0].ipAddress" \
+       --output tsv)
+
+    if echo "$HOST" | grep -Eq "^([0-9]{1,3}[\.]){3}[0-9]{1,3}$"; then
+        break
+    fi
+    if [ "$LOOP_COUNTER" = "10" ]; then
```

Comment:
The counter is never going to go past 10 :)

Reply:
:facepalm:

Reply:
I'm an idiot :crying_cat_face: